cluster/src/main/java/org/apache/iotdb/cluster/config/ClusterDescriptor.java

```diff
@@ -204,4 +206,53 @@ private void loadProps() {
     }
     return urlList;
   }
+
+  public void loadHotModifiedProps() throws QueryProcessException {
+    String url = getPropsUrl();
+    if (url == null) {
+      return;
+    }
+    try (InputStream inputStream = new FileInputStream(new File(url))) {
+      logger.info("Start to reload config file {}", url);
+      Properties properties = new Properties();
+      properties.load(inputStream);
+      loadHotModifiedProps(properties, false);
+    } catch (Exception e) {
+      logger.warn("Fail to reload config file {}", url, e);
+      throw new QueryProcessException(
+          String.format("Fail to reload config file %s because %s", url, e.getMessage()));
+    }
+  }
+
+  public void loadHotModifiedProps(Properties properties, boolean toCheckProperties)
+      throws QueryProcessException {
+    if (toCheckProperties && !checkProperties(properties)) {
+      throw new QueryProcessException(
+          "Failed to load configuration from properties because some are missing locally.");
+    }
+    // TODO implement some hot modified properties to load
+    logger.info("Set cluster configuration {}", properties);
+  }
+
+  private boolean checkProperties(Properties properties) throws QueryProcessException {
+    String url = getPropsUrl();
+    if (url == null) {
+      return false;
+    }
+    try (InputStream inputStream = new FileInputStream(new File(url))) {
+      Properties localProperties = System.getProperties();
+      localProperties.load(inputStream);
+      Set<String> localPropertyNames = localProperties.stringPropertyNames();
+      for (String propertyName : properties.stringPropertyNames()) {
+        if (!localPropertyNames.contains(propertyName)) {
+          return false;
+        }
+      }
```

Comment:
I am not sure if I got it right, but the parameter `properties` is loaded from `getPropUrl()` and you also load `getPropUrl()` into `localProperties`, so it seems that `properties` must be a subset of `localProperties`.

Reply:
It seems that the check there is unnecessary. I will remove it.